bvm/ballerina-core/src/main/java/org/ballerinalang/model/values/BError.java

```diff
@@ -84,4 +85,8 @@ public Object value() {
     public boolean isFrozen() {
         return true;
     }
+
+    public BError getCause() {
+        return cause;
+    }
```

Comment:
Shall we override the equals method. Without it an unit test case is failing.
Following worked in my local machine.
```java
    @Override
    public boolean equals(Object obj) {
        if (this == obj) {
            return true;
        }

        if (obj instanceof  BError) {
            BError that = (BError) obj;

            boolean isCauseSame = false;
            if (this.cause != null) {
                isCauseSame = this.cause.equals(that.cause);
            } else if (this.cause == null && that.cause == null) {
                isCauseSame = true;
            }
            return this.message.equals(that.message) && this.details.equals(that.details) && isCauseSame;
        }
        return false;
    }
```

Reply:
Thanks, Fixed via edc2257a72440de5a34243906a648875a446eef5